src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -1733,6 +1734,12 @@ void primitive_inst::do_runtime_in_place_concat() {
                                << dep.first->_impl_params->output_layouts[0].to_string() << std::endl;
         ++i;
     }
+
+    // If concat primitive's output shape has changed, reset the allocation flag to allow its predecessors
+    // to call realloc_if_needed for concat during their own realloc_if_needed calls
+    if (prev_concat_layout != concat_layout)
+        concat_inst->_allocation_done_by_other = false;
+
```

Comment:
now it is becoming more complex because we have 1 place to turn this on, 2 places to turn this off.
maybe, can we just reset this flag unconditionally when concat node is processed?